benchmarks/kpi/datamodule.py

```diff
@@ -0,0 +1,127 @@
+from typing import Optional, Sequence
+
+import numpy as np
+import pandas as pd
+from sklearn.pipeline import make_pipeline
+from torch.utils.data import DataLoader
+
+from numalogic.tools.data import TimeseriesDataModule, StreamingDataset
+
+
+class KPIDataModule(TimeseriesDataModule):
+    def __init__(
+        self,
+        data_dir: str,
+        kpi_idx: int,
+        preproc_transforms: Optional[list] = None,
+        split_ratios: Sequence[float] = (0.5, 0.2, 0.3),
+        *args,
+        **kwargs,
+    ):
+        super().__init__(data=None, *args, **kwargs)
+
+        if len(split_ratios) != 3 or sum(split_ratios) != 1.0:
+            raise ValueError("Sum of all the 3 ratios should be 1.0")
+
+        self.split_ratios = split_ratios
+        self.data_dir = data_dir
+        self.kpi_idx = kpi_idx
+        if preproc_transforms:
+            if len(preproc_transforms) > 1:
+                self.transforms = make_pipeline(preproc_transforms)
+            else:
+                self.transforms = preproc_transforms[0]
+        else:
+            self.transforms = None
+
+        self.train_dataset = None
+        self.val_dataset = None
+        self.test_dataset = None
+
+        self._train_labels = None
+        self._val_labels = None
+        self._test_labels = None
+
+        self.unique_kpis = None
+
+        self._kpi_df = self.get_kpi_df()
+
+    def _preprocess(self, df: pd.DataFrame):
```

Comment:
What's the expected structure within df?

Reply:
Added in the docstrings